### PR TITLE
fix(dav): crtime parsing and invalid date handling

### PR DESCRIPTION
**Pull request by skjnldsv**
See https://github.com/nextcloud/server/pull/49225

### CODE REVIEW

**Comment by skjnldsv** [comment]
Added tests

**Review by Pytal — APPROVED** [review]
Other than the nit

File: lib/dav/dav.ts [hunk 1]
```diff
@@ -174,10 +175,14 @@ export const davResultToNode = function(node: FileStat, filesRoot = davRootPath,
 	const owner = String(props?.['owner-id'] || userId)
 	const id = props.fileid || 0
 
+	const mtime = new Date(Date.parse(node.lastmod))
+	const crtime = new Date(Date.parse(props.creationdate))
```
[the remark this answers is not in the document]
Comment (susnux) [reply lib/dav/dav.ts:179]: Both is non standard, the Javascript standard only defines the iso format, everything else is implementation dependant. 

But at least Date.parse supports HTTP date format on all supported browsers.

[the remark this answers is not in the document]
Comment (skjnldsv) [reply lib/dav/dav.ts:179]: So, we keep parse or not ?

[the remark this answers is not in the document]
Comment (Pytal) [reply lib/dav/dav.ts:179]: Fine with either as the date string is parsed "using the same algorithm implemented by Date.parse()" https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#datestring

[the remark this answers is not in the document]
Comment (susnux) [reply lib/dav/dav.ts:179]: IIRC safari has some differences, but yes I am also fine with both :man_shrugging:

[the remark this answers is not in the document]
Comment (skjnldsv) [reply lib/dav/dav.ts:179]: Then me lazy, I'll let this as is :+1:

**Pull request merged**